core/src/main/java/com/onelogin/saml2/authn/SamlResponse.java

```diff
@@ -98,12 +124,7 @@
      *
 	 */
 	public SamlResponse(Saml2Settings settings, HttpRequest request) throws XPathExpressionException, ParserConfigurationException, SAXException, IOException, SettingsException, ValidationError {
-		this.settings = settings;
-
-		if (request != null) {
-			currentUrl = request.getRequestURL();
-			loadXmlFromBase64(request.getParameter("SAMLResponse"));
-		}
+		this(settings, request.getRequestURL(), request.getParameter("SAMLResponse"));
```

Comment:
Why are you removing the null check?

Reply:
Because the call to another constructor has to be the first thing in a constructor, so 
```java
if (request != null) {
  this(...)
}
```
is not permitted in Java.